src/log-store/src/kafka/log_store.rs

```diff
@@ -137,8 +143,8 @@ impl LogStore for KafkaLogStore {
             return Ok(AppendBatchResponse::default());
         }
 
-        // Groups entries by region id and pushes them to an associated record producer.
-        let mut producers = HashMap::with_capacity(entries.len());
+        let mut region_grouped_records: HashMap<RegionId, Vec<_>> = HashMap::new();
+        let mut region_producers: HashMap<RegionId, OrderedBatchProducerRef> = HashMap::new();
```

Comment:
Is it possible to `merge region_grouped_records` and `region_producers`, using only one hashmap?